Source/WebKit/WebProcess/Network/WebLoaderStrategy.cpp

```diff
@@ -361,6 +364,20 @@ static void addParametersShared(const LocalFrame* frame, NetworkResourceLoadPara
 
     if (isMainFrameNavigation)
         parameters.linkPreconnectEarlyHintsEnabled = mainFrame->settings().linkPreconnectEarlyHintsEnabled();
+    else if (RefPtr localMainFrame = dynamicDowncast<LocalFrame>(mainFrame)) {
+        if (auto* document = frame->document(); localMainFrame->document()
+            && isRequestCrossOrigin(&localMainFrame->document()->securityOrigin(), parameters.request.url(), { })
+            && document) {
+
+            std::optional<SecurityOriginData> securityOrigin;
+            if (cachedResource && cachedResource->type() == CachedResource::Type::MainResource)
+                securityOrigin = SecurityOriginData::fromURL(parameters.request.url());
+            parameters.isPermissionPolicyPrivateTokenAllowed = isFeaturePolicyAllowedByDocumentAndAllOwners(FeaturePolicy::Type::PrivateToken, *document, LogFeaturePolicyFailure::No, securityOrigin);
```

Comment:
I would tend to do this computation at call site in CachedResourceLoader/CachedResource and add a ResourceLoaderOptions.isPrivacyTokenUsageAllowed.

The service worker integration would also be interesting to study.
If there is a call from a third party iframe that is allowed to use privacy token, it seems the intercepted request by the service worker without modification should also be allowed to use privacy token.
In that case, we need to send that information to the service worker.

Reply:
Do we have a request representation that goes through service workers and ends up in CFNetwork? It seems setting a bit on that that private-token access is enabled for this request should suffice.

(Now once we have site isolation we might have to think harder about this as we could/should presumably constrain cross-site processes better when they don't have certain permissions.)

Reply:
Thanks for these reviews! I moved the computation into FrameLoader and added the result into ResourceRequest. Using ResourceRequest seems to transparently pass the information through a ServiceWorker, too.